src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/AsyncGoalEventService.kt

```diff
@@ -23,18 +23,45 @@ class AsyncGoalEventService(
   override fun goalCreated(prisonNumber: String, createdGoal: Goal) {
     runBlocking {
       log.debug { "Goal created event for prisoner [$prisonNumber]" }
-      val timelineEvent = timelineEventFactory.goalCreatedTimelineEvent(createdGoal)
-      launch { timelineService.recordTimelineEvent(prisonNumber, timelineEvent) }
-      launch { telemetryService.trackGoalCreateEvent(createdGoal) }
+      launch { recordCreateGoalTimelineEvent(prisonNumber = prisonNumber, createdGoal = createdGoal) }
+      launch { trackGoalCreatedTelemetryEvent(createdGoal = createdGoal) }
     }
   }
 
   override fun goalUpdated(prisonNumber: String, previousGoal: Goal, updatedGoal: Goal) {
     runBlocking {
       log.debug { "Goal updated event for prisoner [$prisonNumber]" }
-      val timelineEvents = timelineEventFactory.goalUpdatedEvents(previousGoal = previousGoal, updatedGoal = updatedGoal)
-      launch { timelineService.recordTimelineEvents(prisonNumber, timelineEvents) }
-      launch { telemetryService.trackGoalUpdateEvent(updatedGoal) }
+
+      launch {
+        recordUpdateGoalTimelineEvents(
+          prisonNumber = prisonNumber,
+          previousGoal = previousGoal,
+          updatedGoal = updatedGoal,
+        )
+      }
+      launch { trackGoalUpdatedTelemetryEvents(previousGoal = previousGoal, updatedGoal = updatedGoal) }
     }
   }
+
+  private fun recordCreateGoalTimelineEvent(prisonNumber: String, createdGoal: Goal) {
+    timelineService.recordTimelineEvent(
+      prisonNumber,
+      timelineEventFactory.goalCreatedTimelineEvent(createdGoal),
+    )
+  }
+
+  private fun recordUpdateGoalTimelineEvents(prisonNumber: String, previousGoal: Goal, updatedGoal: Goal) {
```

Comment:
This looks good and nice and simple (which I like!).

Sorry to nitpick, but I think these timeline methods should be recordGoalCreated/UpdatedTimelineEvent(s) (i.e. past tense as well).

Reply:
I agree - good spot. I'll change them in a sec .... 👍

Reply:
Done 👍